homeassistant/components/holiday/config_flow.py

```diff
@@ -0,0 +1,103 @@
+"""Config flow for Holiday integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from babel import Locale
+from holidays import list_supported_countries
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_COUNTRY
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.selector import (
+    CountrySelector,
+    CountrySelectorConfig,
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+)
+
+from .const import CONF_PROVINCE, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SUPPORTED_COUNTRIES = list_supported_countries(include_aliases=False)
+
+
+class HolidayConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Holiday."""
+
+    VERSION = 1
+
+    data: dict[str, Any] = {}
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the initial step."""
+        if user_input is not None:
+            self.data = user_input
+
+            selected_country = self.data[CONF_COUNTRY]
+
+            if SUPPORTED_COUNTRIES[selected_country]:
+                return await self.async_step_province()
+
+            self._async_abort_entries_match({CONF_COUNTRY: user_input[CONF_COUNTRY]})
+
+            locale = Locale(self.hass.config.language)
+            title = locale.territories[selected_country]
+            return self.async_create_entry(title=title, data=self.data)
+
+        user_schema = vol.Schema(
+            {
+                vol.Required(
+                    CONF_COUNTRY, default=self.hass.config.country
+                ): CountrySelector(
+                    CountrySelectorConfig(
+                        countries=list(SUPPORTED_COUNTRIES),
+                    )
+                ),
+            }
+        )
+
+        return self.async_show_form(step_id="user", data_schema=user_schema)
+
+    async def async_step_province(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle the province step."""
+        if user_input is not None:
+            combined_input: dict[str, Any] = {**self.data, **user_input}
+
+            country = combined_input[CONF_COUNTRY]
+            province = combined_input.get(CONF_PROVINCE)
+
+            self._async_abort_entries_match(
+                {
+                    CONF_COUNTRY: country,
+                    CONF_PROVINCE: province,
+                }
+            )
+
+            locale = Locale(self.hass.config.language)
+            province_str = f", {province}" if province else ""
+            name = f"{locale.territories[country]}{province_str}"
```

Comment:
I think there should be a separator between country and province?

Reply:
There is a comma, see line 86.